elastic/kibana
# [Response Ops][Alerting] Defining default action params on connector type

Towards https://github.com/elastic/kibana/issues/90931

## Summary

Decoupling default action parameters so that we can move the connector types UI code to the `stack_connectors` plugin. Currently Pagerduty and XMatters connectors define some default action parameters that are auto-populated for users when they add actions to their rules. These definitions were stored in a separate file and the way they were passed to the `ActionTypeForm` component made it difficult to untangle the connector code from the `triggers_actions_ui` framework code. This PR moves those definitions to the connector type definition.

## To Verify

1. Create a Pagerduty connector
2. Create a rule and add a Pagerduty action. See that the dedupeKey and event action on the action form are auto-populated. Switch the action group to Recovered and see that the event action auto-changes from Trigger to Resolve
3. Create an Xmatters connector
4. Create a rule and add an Xmatters action. Save the rule and then inspect the rule saved object. See that the Xmatters action is auto-populated with parameters.
